incubator/guides-restructured-text/src/RestructuredText/Parser/LinesIterator.php

```diff
@@ -28,18 +31,32 @@
 class LinesIterator implements Iterator
 {
     /** @var string[] */
-    private $lines = [];
-
-    /** @var int */
-    private $position = 0;
+    private array $lines = [];
+    private int $position = 0;
 
     public function load(string $document): void
     {
         $document = trim($this->prepareDocument($document));
         $this->lines = explode("\n", $document);
+
+        foreach ($this->lines as $key => $line) {
```

Comment:
Should this not be part of the prepare document function?

Reply:
The prepareDocument function treats the contents as a whole, and before I can remove tab indentation it needs to be split into lines. A search and replace on all tabs could be detrimental.. If I were to move this into the prepareDocument method, then its output would change to lines instead of text. Is that desired?

Reply:
It makes sense to do it by line. As we should only replace it at the start of the line. Leave it where it is right now 👍